pkg/types/gcp/validation/machinepool.go

```diff
@@ -80,3 +84,45 @@ func ValidateDefaultDiskType(p *gcp.MachinePool, fldPath *field.Path) field.Erro
 
 	return allErrs
 }
+
+func ValidateKMSKey(platform *gcp.Platform, p *types.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if p != nil && p.Platform.GCP != nil && p.Platform.GCP.OSDisk.EncryptionKey != nil {
+		kmsClient, err := kms.NewKeyManagementClient(context.Background())
+		if err != nil {
+			return append(allErrs, field.InternalError(fldPath.Child("encryptionKey"), err))
+		}
+		defer kmsClient.Close()
+
+		project := p.Platform.GCP.OSDisk.EncryptionKey.KMSKey.ProjectID
+		if project == "" {
+			project = platform.ProjectID
+		}
+		location := p.Platform.GCP.OSDisk.EncryptionKey.KMSKey.Location
+		if location == "" {
+			location = platform.Region
+		}
+
+		req := &kmspb.ListKeyRingsRequest{
+			Parent: fmt.Sprintf("projects/%s/locations/%s", project, location),
+		}
+
+		it := kmsClient.ListKeyRings(context.Background(), req)
+		for {
+
+			resp, err := it.Next()
+			if err == iterator.Done {
+				break
+			}
+			if err != nil {
+				return append(allErrs, field.InternalError(fldPath.Child("encryptionKey"), err))
+			}
+			if resp.Name == p.Platform.GCP.OSDisk.EncryptionKey.KMSKey.Name {
+				return allErrs
+			}
+		}
+	}
+
+	return allErrs
```

Comment:
If the iterator goes through all the keys in the keyring and does not find the KMSKey provided by the user, then it appears that we return `field.ErrorList{}`? Shouldn't we be returning an error?